hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -711,7 +711,7 @@ object Simplify {
         && child.typ.key.nonEmpty && canRepartition =>
       TableAggregateByKey(child, expr)
 
-    case TableAggregateByKey(TableKeyBy(child, keys, _), expr) if canRepartition =>
+    case TableAggregateByKey(TableKeyBy(child, keys, false), expr) if canRepartition && ! child.typ.key.startsWith(keys) =>
```

Comment:
If `child.typ.key.startsWith(keys)`, then `TableKeyBy` will ignore the `isSorted` flag, so I don't think you want to match on `isSorted = false`.

Reply:
I want both. If we're lengthening the key without shuffling, we should not rewrite this as a shuffling aggregation.

Reply:
Ah, I get it now. I still think it's clearer matching on `TableKeyBy(child, keys, _)`, since `!x.defininitelyDoesNotShuffle` already implies `isSorted = false`, but it's not a big deal. Approving.

Reply:
Ah, I get it now. I still think it's clearer matching on `TableKeyBy(child, keys, _)`, since `!x.definitelyDoesNotShuffle` already implies that `x.isSorted = false`, but it's not a big deal. Approving.

Reply:
👍